src/rdf4cpp/bnode_mngt/reference_backends/generator/IncreasingIdGenerator.cpp

```diff
@@ -11,20 +11,21 @@ namespace generator_detail {
 static constexpr size_t max_generated_id_size = std::numeric_limits<size_t>::digits10;
 } //namespace generator_detail
 
-IncreasingIdGenerator::IncreasingIdGenerator(std::string prefix, size_t const initial_value) noexcept : prefix{std::move(prefix)},
-                                                                                                        counter{initial_value} {
+IncreasingIdGenerator::IncreasingIdGenerator(std::string prefix, size_t const initial_value) noexcept : prefix_{std::move(prefix)},
+                                                                                                        counter_{initial_value} {
 }
 
-size_t IncreasingIdGenerator::max_generated_id_size() const noexcept {
-    return generator_detail::max_generated_id_size + this->prefix.size();
-}
+BlankNode IncreasingIdGenerator::generate(storage::DynNodeStoragePtr node_storage) noexcept {
+    static thread_local std::string buf;
 
-char *IncreasingIdGenerator::generate_to_buf(char *buf) {
-    buf = std::copy(this->prefix.begin(), this->prefix.end(), buf);
+    buf.clear();
+    buf.resize(prefix_.size() + generator_detail::max_generated_id_size);
```

Comment:
is this to avoid constructing a new string every time this function is called?

Reply:
yes, this way we only have one heap allocation per thread instead of one per generated node